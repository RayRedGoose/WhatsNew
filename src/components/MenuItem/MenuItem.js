import React from 'react';
import './MenuItem.scss';
import PropTypes from 'prop-types';

const MenuItem = (props) => {
  const updateActiveTab = (event) => {
    props.activateItem(props.name)
    props.changeContent(props.name)
  }

  const keyPressed = event => {
    if (event.key === "Enter") {
      updateActiveTab(event)
    }
  }

  return (
    <div
      role="button"
      tabIndex="0"
      name={props.name}
      className={props.elementClass}
      onClick={updateActiveTab}
      onKeyPress={keyPressed}>
      <img src={props.iconName} alt={props.text + " icon"}/>
      <h2>{props.text}</h2>
    </div>
  );
}

MenuItem.propTypes = {
  name: PropTypes.string,
  elementClass: PropTypes.string,
  iconName: PropTypes.string,
  text: PropTypes.string,
  activateItem: PropTypes.func,
  changeContent: PropTypes.func,
}

export default MenuItem;
