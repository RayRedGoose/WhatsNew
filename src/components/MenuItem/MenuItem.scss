.menu_item {
  border-top-right-radius: 50px;
  border-bottom-right-radius: 50px;
  cursor: pointer;
  display: flex;
  align-items: center;
  padding: 10px 20px;
  margin-bottom: 20px;
  &:hover,
  &:focus {
    border: 2px solid #f0aa34;
  }
  &:active,
  &.active-tab {
    background-color: #f0aa34;
  }
  img {
    width: 30px;
    margin-right: 20px;
  }
  h2 {
    color: #fff;
    font-size: 22px;
    font-weight: 400;
  }
}
