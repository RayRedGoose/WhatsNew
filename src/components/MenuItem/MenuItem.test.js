import React from 'react';
import { shallow } from 'enzyme';
import MenuItem from './MenuItem';

describe('MenuItem', () => {
  let element,
  activateItem = jest.fn(),
  changeContent = jest.fn()

  beforeEach(() => {
    element = shallow(<MenuItem
      key = "1"
      elementClass="menu_item active-tab"
      name="Health"
      iconName='./Health.svg'
      text='Health'
      activateItem = {activateItem}
      changeContent = {changeContent}
    />);
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(element).toMatchSnapshot();
  });

  it('should call the activateItem prop when clicked', () => {
    element.simulate('click');
    expect(activateItem).toHaveBeenCalledWith('Health');
  });

  it('should call the changeContent prop when clicked', () => {
    element.simulate('click');
    expect(changeContent).toHaveBeenCalledWith('Health');
  });

  it('should call the activateItem prop when enter is key', () => {
    element.simulate('keypress', {key: 'Enter'});
    expect(activateItem).toHaveBeenCalledWith('Health');
  });

  it('should call the changeContent when enter is key', () => {
    element.simulate('keypress', {key: 'Enter'});
    expect(changeContent).toHaveBeenCalledWith('Health');
  });

})
