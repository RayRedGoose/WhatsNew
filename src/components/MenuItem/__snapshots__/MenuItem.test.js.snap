// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should match the snapshot with all data passed in correctly 1`] = `
<div
  className="menu_item active-tab"
  name="Health"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <img
    alt="Health icon"
    src="./Health.svg"
  />
  <h2>
    Health
  </h2>
</div>
`;
